src/main/java/seedu/duke/models/schema/Student.java

```diff
@@ -203,4 +214,136 @@ public ModuleList getModulesPlanned() {
     public void printSchedule(){
         this.schedule.printMainModuleList();
     }
+
+    public void setCurrentSemesterModules() {
+        try {
+            int[] yearAndSem = Parser.parseStudentYear(year);
+            System.out.println(Arrays.toString(yearAndSem));
+            int currSem = ((yearAndSem[0] - 1) * 2) + yearAndSem[1];
+            int[] modulesPerSem = schedule.getModulesPerSem();
+            System.out.println(Arrays.toString(modulesPerSem));
+            ModuleList modulesPlanned = schedule.getModulesPlanned();
+            // schedule.printMainModuleList();
+            // System.out.println("-------------------------------");
+            // printSchedule();
+            int numberOfModulesInCurrSem = modulesPerSem[currSem - 1];
+            int numberOfModulesCleared = 0;
+            for (int i = 0; i < currSem - 1; i++) {
+                numberOfModulesCleared += modulesPerSem[i];
+            }
+            int startIndex = numberOfModulesCleared;
+            int endIndex = startIndex + numberOfModulesInCurrSem;
+            ArrayList<Module> modulesInSchedule = modulesPlanned.getMainModuleList();
```

Comment:
Specifying what the code is supposed to do can help other developers compare your comments to the implementation to verify whether your implementation is correct! Explaining the rationale for your current implementation would also add in their understanding!